frontend/src/utils/api.js

```diff
@@ -40,30 +40,58 @@ function getEventAttendees(id: string) {
   return fetch(`${API_URL}/events/${id}/attendees?key=${getKey()}`).then(res => res.json())
 }
 
-function addInterviewSchedule(file: File) {
-  var reader = new FileReader()
-  var scheduleString = ''
-  reader.onload = function(e) {
-    scheduleString = reader.result
-    fetch(`${API_URL}/schedule/upload/?key=${getKey()}`, {
-      method: 'POST',
-      headers: {
-        'content-type': 'application/json'
-      },
-      body: JSON.stringify({ schedule: scheduleString })
-    })
-      .then(res => res.json())
-      .then(success => console.log(success))
-      .catch(error => console.log(error))
-  }
-
-  reader.readAsText(file)
+async function addInterviewerSchedules(file: File) {
+  const scheduleString = await readUploadedFile(file)
+  return fetch(`${API_URL}/schedule/uploadInterviewers/?key=${getKey()}`, {
+    method: 'POST',
+    headers: {
+      'content-type': 'application/json'
+    },
+    body: JSON.stringify({ data: scheduleString })
+  }).then(res => res.json())
+}
+
+async function addCandidateSchedules(file: File) {
+  const scheduleString = await readUploadedFile(file)
+  return fetch(`${API_URL}/schedule/uploadCandidates/?key=${getKey()}`, {
+    method: 'POST',
+    headers: {
+      'content-type': 'application/json'
+    },
+    body: JSON.stringify({ data: scheduleString })
+  }).then(res => res.json())
+}
+
+const readUploadedFile = inputFile => {
+  const reader = new FileReader()
+
+  return new Promise((resolve, reject) => {
+    reader.onerror = () => {
+      reader.abort()
+      reject(new DOMException('Problem parsing input file.'))
```

Comment:
TODO: Change from `DOMException` to something more closely matching the error happening.